src/bin/ion/commands/beta/generate/templates/rust/struct.templ

```diff
@@ -29,106 +31,52 @@ pub mod  {{ target_kind_name | snake }} {
 
         pub fn read_from(reader: &mut Reader) -> SerdeResult<Self> {
             let mut abstract_data_type = {{ target_kind_name }}::default();
-            {% if abstract_data_type == "Value"%}
-                abstract_data_type.value = {% if fields[0].value_type | is_built_in_type == false %}
-                                                {{ fields[0].value_type }}::read_from(reader)?;
-                                            {% else %}
-                                                reader.read_{% if fields[0].isl_type_name == "symbol" %}symbol()?.text().unwrap(){% else %}{{ fields[0].value_type | replace(from="string", to ="str") }}()?{% endif %}{% if fields[0].value_type | lower == "string" %} .to_string() {% endif %};
-                                            {% endif %}
-            {% elif abstract_data_type is object and abstract_data_type is containing("Structure") %}
-                reader.step_in()?;
-                while reader.next()? != StreamItem::Nothing {
-                    if let Some(field_name) = reader.field_name()?.text() {
-                        match field_name {
-                            {% for field in fields -%}
-                                {% if field.value_type | is_built_in_type == false %}
-                                     "{{ field.name }}" => { abstract_data_type.{{ field.name | snake }} = {{ field.value_type }}::read_from(reader)?; }
+
+            reader.step_in()?;
+            while reader.next()? != StreamItem::Nothing {
+                if let Some(field_name) = reader.field_name()?.text() {
+                    match field_name {
+                        {% for field in fields -%}
+                            {% if field.value_type | is_built_in_type == false %}
+                                {% if field.value_type is containing("Vec") %}
+                                     "{{ field.name }}" => { {{ util_macros::read_as_sequence(field=field) }} }
```

Comment:
🗺️ PR tour) This statement is modified to read fields that are of type `Vec` using util_macros `write_as_sequence`. Previously it was not possible for a field in struct to be of sequence type(e.g. It would be inside some nested type `NestedtypeX`). But with this change it is possible to have fields with sequence type (e.g. `Vec<i64>`).